src/reason-parser/reason_pprint_ast.ml

```diff
@@ -4117,15 +4160,20 @@ let printer = object(self:'self)
                               atom "=";
                               (label (self#longident_loc lid) (atom "."))])
                    (self#formatNonSequencyExpression e)
+           | Pexp_apply (eFun, _) ->
+             let lhs = (makeList [atom lbl; atom "="]) in
+             let rhs = (match printedStringAndFixityExpr eFun with
```

Comment:
It looks a bit strange to have "implementation details" about token precedence leaking into the formatting logic of jsx.
Idea: what if we make `.`, `#`, `->` etc. `Simple` ?

Given:
- extreme high precedence
- `^.` already emits as `POSTFIXOP "^" DOT` in the lexer
- `->`, `#` now have the same "access"-like precedence as `.`

This makes them `Simple`, no? If the unparsing emits them as `Simple`, then we could avoid the leakage of precedence details in other parts of the printer.
Thoughts?

Reply:
Also can we add extra unit tests to verify the printing output?

```reason
<div bar=f->g->z />;
<div bar=foo##bar />;
```